src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -517,20 +517,33 @@ public void AddMetadataReference(MetadataReference reference, AssemblyIdentity i
             }
         }
 
-        private static readonly string s_GAC = $"{Path.DirectorySeparatorChar}GAC_MSIL{Path.DirectorySeparatorChar}";
-        private static readonly string s_ALTGAC = $"{Path.AltDirectorySeparatorChar}GAC_MSIL{Path.AltDirectorySeparatorChar}";
-
         private bool IsInGAC(string filePath)
         {
-            return filePath.Contains(s_GAC) || filePath.Contains(s_ALTGAC);
+            return GlobalAssemblyCacheLocation.RootLocations.Any(gloc => FilePathUtilities.IsNestedPath(gloc, filePath));
```

Comment:
Thsi code needs some cleanup -- there is also a helper in Workspaces that's not used anymore:
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Utilities/ReferencePathUtilities_Desktop.cs,9678a838cc386bc4,